adr-0059: Welkin to Consist of Public Open Source Code and Proprietar… (#1032)
elastisys/welkin: docs/adr/0059-welkin-to-consist-public-open-source-code-and-proprietary-documentation.md

# Welkin to Consist of Public Open Source Code and Proprietary Documentation

- Status: Accepted
- Deciders: Management Team
- Date: 2025-01-20

## Context and Problem Statement

Welkin’s current policy, as defined in ADR-0055, is to maintain a mix of public and private open-source components for its product. This approach aimed to balance openness with control over sensitive intellectual property.

> However, a critical challenge has emerged. Infrastructure operators, who are key stakeholders, require access to the codebase to understand, troubleshoot, and maintain critical systems. Restricting access to portions of the code limits their ability to operate effectively and creates friction in their workflows.

Reviewer: ```suggestion
However, new information has emerged.
Essential and important entities (as defined in NIS2), who are key stakeholders, require access to the codebase to understand, troubleshoot, and maintain critical systems.
Restricting access to portions of the code limits their ability to operate effectively and creates friction in their workflows.
```

Reviewer 2: Looks great to me! :)

Additionally, by maintaining a private portion of the open-source code, we introduce unnecessary complexity into our development and operational processes. Ensuring parity between public and private repositories adds overhead and detracts from our goals of transparency and ease of use.

> Given these factors, should Welkin move to a model of fully public open-source code while keeping documentation proprietary.

Reviewer: ```suggestion
Given these factors, should Welkin move to a model of fully public open-source code while keeping documentation proprietary?
```

Reviewer 2: Thumbs up, this is formulated as a question.

## Decision Drivers

- Elastisys wants to protect Welkin users' open source freedom
- Elastisys wants to offer product support customers good value for their money, and incentivise these to keep paying Elastisys, not only for support but also for access to future Welkin development, security patches, etc.
- Product support customers should have ability to adapt Welkin if needed
- Elastisys was founded on a bedrock on knowledge sharing and openness, and wants that to be reflected in the products and services it develops
- Elastisys needs to protect its intellectual property

## Considered Options

> 1. Yes, move to a model of fully public open-source code while keeping documentation proprietary.

Reviewer: ```suggestion
1. Yes, move to a model of fully public open-source code.
```

Let's focus on the positive only.

> 1. No, keep decision made in ADR0055

Reviewer: ```suggestion
1. No, keep decision made in ADR-0055
```

Reviewer 2: Nice. Link, perhaps, too?

Author: Nice catch
I thought I added it :thinking: 
Added it just now

## Decision Outcome

> Option 1: Move to a model of fully public open-source code while keeping documentation proprietary.

Reviewer: ```suggestion
Option 1: Move to a model of fully public open-source code.
```
